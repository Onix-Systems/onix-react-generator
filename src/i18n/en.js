export default {
  welcome: 'Try to build something great',
  home: 'Home',
  about: 'About',
  aboutPage: 'About page',
  homePage: 'Home page',
  newJoke: 'New joke',
};
