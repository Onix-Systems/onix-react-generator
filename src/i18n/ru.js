export default {
  welcome: 'Нас ждут великие дела',
  home: 'Главная',
  about: 'О нас',
  aboutPage: 'О нас',
  homePage: 'Главная страница',
  newJoke: 'Новая шутка',
};
