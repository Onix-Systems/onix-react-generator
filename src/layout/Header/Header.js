import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import * as translateActions from '../../store/translates/actions';
import Title from '../../components/Title/Title';
import styles from './sass/Header.module.scss';

function Header({
  home, about, actions, message, locale
}) {
  return (
    <header className={styles.appHeader}>
      <Title text={message} />
      <Link to="/" className={styles.headerLink}>
        {home}
      </Link>
      <Link to="/about" className={styles.headerLink}>
        {about}
      </Link>
      <button
        className={styles.toggleLanguage}
        data-testid="toggle-language-button"
        type="button"
        onClick={actions.changeLanguage}
      >
        {locale}
      </button>
    </header>
  );
}

Header.propTypes = {
  home: PropTypes.string.isRequired,
  about: PropTypes.string.isRequired,
  actions: PropTypes.shape({
    changeLanguage: PropTypes.func.isRequired
  }).isRequired,
  message: PropTypes.string.isRequired,
  locale: PropTypes.string.isRequired,
};

const mapStateToProps = (state) => ({
  locale: state.translates.locale
});

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators(translateActions, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
