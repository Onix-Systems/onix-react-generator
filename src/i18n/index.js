import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import en from './en';
import ru from './ru';

export const resources = {
  ru: { translation: ru },
  en: { translation: en },
};

export const appLocales = Object.keys(resources);

const customDetector = {
  name: 'languageDetector',
  lookup() {
    let appLanguage = appLocales[0];
    const locale = window.navigator.language;
    const localeShort = locale.split(/_|-/)[0];
    const isPresent = appLocales.find((item) => item === localeShort);
    const storeLang = localStorage.getItem('i18nextLng');
    if (storeLang) {
      appLanguage = storeLang;
    } else if (isPresent) {
      appLanguage = localeShort;
    }
    return appLanguage;
  },
};

const detectorOptions = {
  order: ['languageDetector', 'localStorage'],
};

const languageDetector = new LanguageDetector();
languageDetector.addDetector(customDetector);

i18n
  .use(languageDetector)
  .use(initReactI18next)
  .init({
    resources,
    fallbackLng: appLocales,
    react: {
      useSuspense: true,
    },
    detection: detectorOptions,
  });

export default i18n;
