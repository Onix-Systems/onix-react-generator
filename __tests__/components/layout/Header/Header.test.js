import React from 'react';
import userEvent from '@testing-library/user-event';
import i18n from 'i18next';
import { render } from '../../../settings/test-utils';
import Header from '../../../../src/layout/Header/Header';

describe('<Header />', () => {
  let wrapper;
  const mockText = 'mock text';
  beforeEach(() => {
    wrapper = render(
      <Header
        home={mockText}
        about={mockText}
        message={mockText}
      />
    );
  });

  test('should render Header component', () => {
    const { asFragment } = wrapper;
    expect(asFragment(wrapper)).toMatchSnapshot();
  });

  test('should toggle language', (done) => {
    const { getByTestId } = wrapper;
    const Button = getByTestId('toggle-language-button');
    expect(Button).toHaveTextContent(i18n.language);
    userEvent.click(Button);
    expect(Button).toHaveTextContent(i18n.language);
    done();
  });
});
