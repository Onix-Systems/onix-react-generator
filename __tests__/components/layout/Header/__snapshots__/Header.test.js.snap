// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render Header component 1`] = `
<DocumentFragment>
  <header
    class="appHeader"
  >
    <h1
      class="appTitle"
    >
      mock text
    </h1>
    <a
      class="headerLink"
      href="/"
    >
      mock text
    </a>
    <a
      class="headerLink"
      href="/about"
    >
      mock text
    </a>
    <button
      class="toggleLanguage"
      data-testid="toggle-language-button"
      type="button"
    >
      en
    </button>
  </header>
</DocumentFragment>
`;
